waku/factory/internal_config.nim

```diff
@@ -31,24 +31,12 @@ proc enrConfiguration*(conf: WakuNodeConf, netConfig: NetConfig, key: crypto.Pri
 
   enrBuilder.withMultiaddrs(netConfig.enrMultiaddrs)
 
-  let topics =
-    if conf.pubsubTopics.len > 0 or conf.contentTopics.len > 0:
-      let shardsRes = conf.contentTopics.mapIt(getShard(it))
-      for res in shardsRes:
-        if res.isErr():
-          error "failed to shard content topic", error=res.error
-          return err($res.error)
-
-      let shards = shardsRes.mapIt(it.get())
-
-      conf.pubsubTopics & shards
-    else:
-      conf.topics
-
-  let addShardedTopics = enrBuilder.withShardedTopics(topics)
-  if addShardedTopics.isErr():
-      error "failed to add sharded topics to ENR", error=addShardedTopics.error
-      return err($addShardedTopics.error)
+  let shards: seq[uint16] =
+    if conf.shards.len == 0: toSeq(0..8).mapIt(uint16(it))
+    else: toSeq(conf.shards.mapIt(uint16(it)))
+
+  enrBuilder.withWakuRelaySharding(RelayShards(clusterId:uint16(conf.clusterId), shardIds: shards)).isOkOr():
```

Comment:
imho the ENR doesnt require knowledge of topics. just `cluster-id` and `shards`

Reply:
The problem is that pubsub topics in the config can be static shards. Maybe we could disallow that somehow?

Reply:
mm i see. in that case we need a flag that is static/auto?